### PR TITLE
ROU-10963: Fixed tab navigation inside sidebar.

### PR DESCRIPTION
**Pull request by joselrio**
This PR will fix Tab navigation inside sidebar.

### What was happening

- Sidebar elements was being ignored at tab navigation.

### What was done

- Fixed the issue that causing it by updating the query of focusable elements in order to include hidden element to the tab navigation as well.

> **After fix:**
> after-fix2

### CODE REVIEW

**Review by BenOsodrac — APPROVED** [review]

**Review by JoaoFerreira-FrontEnd — APPROVED** [review]

File: src/scripts/OSFramework/OSUI/Behaviors/FocusTrap.ts [hunk 1]
```diff
@@ -14,10 +15,11 @@ namespace OSFramework.OSUI.Behaviors {
 	 * @class FocusTrap
 	 */
 	export class FocusTrap {
+		private _canContainOtherPatterns = false;
 		private _firstFocusableElement: HTMLElement;
 		private _focusBottomCallback: GlobalCallbacks.Generic;
 		private _focusTopCallback: GlobalCallbacks.Generic;
-		private _focusableElements: HTMLElement[];
+		private _focusableElements: HTMLElement[] = [];
```
Comment (rugoncalves) [inline src/scripts/OSFramework/OSUI/Behaviors/FocusTrap.ts:22]: Why are you initializing the array here, and not in the constructor?

File: src/scripts/OSFramework/OSUI/Behaviors/FocusTrap.ts [hunk 2]
```diff
@@ -115,15 +120,35 @@ namespace OSFramework.OSUI.Behaviors {
...
+			const possibleFocusableElements = Helper.Dom.GetFocusableElements(
+				this._targetElement,
+				includeTabIndexHidden
+			);
+
+			// Ensure we get the proper list of focusable elements
+			for (const element of possibleFocusableElements) {
+				// Check if the element is a child of the given targetElement
+				if (
+					element.closest(GlobalEnum.DataBlocksTag.DataBlock) === this._targetElement ||
```
Comment (rugoncalves) [inline src/scripts/OSFramework/OSUI/Behaviors/FocusTrap.ts:137]: Do you think it's worth create a `const` with `element.closest(GlobalEnum.DataBlocksTag.DataBlock)`? Not so much performance, but rather readability of the code.

Comment (joselrio) [reply src/scripts/OSFramework/OSUI/Behaviors/FocusTrap.ts:137]: yup, makes sense... ;)

Comment (joselrio) [reply src/scripts/OSFramework/OSUI/Behaviors/FocusTrap.ts]: Done!

File: src/scripts/OSFramework/OSUI/Behaviors/FocusTrap.ts (hunk 1 above)
Comment (joselrio) [reply src/scripts/OSFramework/OSUI/Behaviors/FocusTrap.ts]: Done!

**Review by JoaoFerreira-FrontEnd — APPROVED** [review]

**Review by rugoncalves — APPROVED** [review]

**Review by JoaoFerreira-FrontEnd — APPROVED** [review]

**Pull request merged**